### ISSUE TITLE
[Feature Request] Optimize scheduling based on Card Types (not Note Types)

### ISSUE THREAD

**Issue by Lila-Blume** [open-spaced-repetition/fsrs4anki#537]
**Which module is related to your feature request?**
Scheduler

I would love to be able to apply different presets to my cards depending on their card types. I use notes with multiple siblings and those vary in their difficulty and at which time in their lifespan I want to see them more often. Right now I tend to aggressively grade the easier types easy later on to space them out more than the harder to recall types.

A similar feature had been suggested before but with tags: https://github.com/open-spaced-repetition/fsrs4anki/issues/380
I agree that it doesn't make sense to use tags for this because cards can have more than one tag. Each card can however only have one single card type, so this might be possible instead.

**Comment by L-M-Sherlock** [comment]
I recommend moving the easier type of cards into another deck. Then, you can apply a different preset to the deck.

**Comment by Lila-Blume** [comment]
I'm talking about "Card Types" vs "Note Types". Notes can be sorted into decks, cards can't. They will always all follow their parent note.
Right now, optimizer presets are applied on a Note Type basis. I'm suggesting that it would be useful to do this on the Card Type basis instead to allow separate scheduling of siblings.

**Comment by L-M-Sherlock** [comment]
I don't understand. Do you mean it's impossible to separate two cards of the same note into two different decks?

**Comment by Vilhelm-Ian** [comment]
he means that  both cards will be scheduled by the same preset even if they are in seperate decks

**Comment by L-M-Sherlock** [comment]
> he means that both cards will be scheduled by the same preset even if they are in separate decks

Is it correct? Should they be scheduled by different presets when the decks have different presets?

**Comment by Vilhelm-Ian** [comment]
he means that  both cards will be scheduled by the same preset even if they are in seperate decks

**Comment by Lila-Blume** [comment]
> I don't understand. Do you mean it's impossible to separate two cards of the same note into two different decks?

Whoah, I'm honestly mind blown right now. I've been using Anki for years and did not know that this was possible because I create my decks from cvs imports and it only gives you the one deck selection on import. You are indeed right, I can move siblings of the same note into separate decks! I did not know that at all. I always thought decks were note specific, and not card specific collections.
Thanks for teaching me something new!

**Comment by Lila-Blume** [comment]
Well, with this knowledge this is indeed already possible by moving the siblings into separate decks and applying different presets to those decks. Thanks for the help!

**Comment by Lila-Blume** [comment]
Oh, actually this doesn't completely clear it up yet:

> > he means that both cards will be scheduled by the same preset even if they are in separate decks
> 
> Is it correct? Should they be scheduled by different presets when the decks have different presets?

Let's say the deck structure looks like this. Each subdeck uses a separate scheduling preset:

- Parent Deck
- - Subdeck Type A
- - Subdeck Type B
- - Subdeck Type C

Now what happens if I want to study them all together from within their parent deck. Are the individual schedules going to be overwritten by the intervals of the parent deck? Or are the cards going to keep their own schedules from their subdecks?

**Comment by Vilhelm-Ian** [comment]
they will be scheduled based on the subdeck in which they lie

**Comment by Lila-Blume** [comment]
Awesome, thank you for the clarification. I felt like I had seen this answered somewhere before but wasn't sure anymore after the comments. That's great, that makes it already fully possible to have siblings run on individual schedules.

**Comment by Lila-Blume** [comment]
I would like to open this again as a feature request. Not instead of what we already have, but additionally. I know the system works well for most users, but there will always be some (like me) who would profit from a different setup.

Let me explain my situation:
I study languages. All of my notes have several different card types that vary a lot in difficulty. I also organize my decks in lots of different sub-chapters, following my lesson plan. For reviewing I use filtered decks that collect everything together, but for the initial learning I use the individual lesson decks (hand in hand with physical books) to learn the notes in order. They build onto each other, so that's necessary.
I used to have this organized with sub-decks for the lessons and filtered decks for different daily study situations. That way the sub-decks were unaffected by burying and could be used again when I needed to review something in context. And the filtered study decks would only gather what wasn't buried on any day so avoided the immediate overlap of siblings after the initial learning.

After our discussion last year, I restructured everything so that the different card types could have individual sub-decks by type and with that individual FSRS parameters. For that to work though, I had to organize my lesson decks in filtered deck instead and here the annoyances with the burying started. I ended up having to disable sibling burying in general because otherwise I can't gather cards in lesson context anymore.
The sub-decks that are only created to get the separate parameters and are never used for studying are also cluttering up the deck list unnecessarily.

So here's my proposal:
Have FSRS options available in the card types window similar to what's in the deck options. They would apply those options by card type instead of decks. Also have a little checkmark on each that says "override deck parameters" that needs to be set for it to be enabled (obviously disabled by default), so that nobody does it on accident. I don't think it would get in the way of most users who have no need for it but it could still be a valuable option for those who are interested.

Is that something that could potentially be implemented or is the way FSRS works now too focused on decks as a structure to be possible?
